### ISSUE TITLE
lots of errors for aarch64-linux-android

### ISSUE THREAD

**Issue by kalmari246** [DoumanAsh/async-timer#5]
got this while compiling for aarch64-linux-android

```rust
error[E0425]: cannot find function `timerfd_create` in crate `libc`
  --> /data/data/com.termux/files/home/.cargo/registry/src/github.com-1ecc6299db9ec823/async-timer-0.6.1/src/oneshot/timer_fd.rs:17:33
   |
17 |         let fd = unsafe { libc::timerfd_create(libc::CLOCK_MONOTONIC, libc::TFD_NONBLOCK) };
   |                                 ^^^^^^^^^^^^^^ help: a constant with a similar name exists: `SYS_timerfd_create`

error[E0425]: cannot find value `TFD_NONBLOCK` in crate `libc`
  --> /data/data/com.termux/files/home/.cargo/registry/src/github.com-1ecc6299db9ec823/async-timer-0.6.1/src/oneshot/timer_fd.rs:17:77
   |
17 |         let fd = unsafe { libc::timerfd_create(libc::CLOCK_MONOTONIC, libc::TFD_NONBLOCK) };
   |                                                                             ^^^^^^^^^^^^ help: a constant with a similar name exists: `EFD_NONBLOCK`

error[E0412]: cannot find type `itimerspec` in crate `libc`
  --> /data/data/com.termux/files/home/.cargo/registry/src/github.com-1ecc6299db9ec823/async-timer-0.6.1/src/oneshot/timer_fd.rs:23:32
   |
23 |     fn set(&self, timer: libc::itimerspec) {
   |                                ^^^^^^^^^^
help: a struct with a similar name exists
   |
23 |     fn set(&self, timer: libc::timespec) {
   |                                ^^^^^^^^
help: possible candidate is found in another module, you can import it into scope
   |
6  | use crate::oneshot::posix::ffi::itimerspec;
   |

error[E0425]: cannot find function `timerfd_settime` in crate `libc`
  --> /data/data/com.termux/files/home/.cargo/registry/src/github.com-1ecc6299db9ec823/async-timer-0.6.1/src/oneshot/timer_fd.rs:24:34
   |
24 |         let ret = unsafe { libc::timerfd_settime(self.0, 0, &timer, ptr::null_mut()) };
   |                                  ^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `SYS_timerfd_settime`

error[E0422]: cannot find struct, variant or union type `itimerspec` in crate `libc`
  --> /data/data/com.termux/files/home/.cargo/registry/src/github.com-1ecc6299db9ec823/async-timer-0.6.1/src/oneshot/timer_fd.rs:74:27
   |
74 |     let new_value = libc::itimerspec {
   |                           ^^^^^^^^^^
help: a struct with a similar name exists
   |
74 |     let new_value = libc::timespec {
   |                           ^^^^^^^^
help: possible candidate is found in another module, you can import it into scope
   |
6  | use crate::oneshot::posix::ffi::itimerspec;
   |
```
lol?

**Comment by DoumanAsh** [comment]
Welp, that's sad.
I think I've been checking android build before....
I'll fix it

**Comment by DoumanAsh** [comment]
ah damn for some fucking reason `timerfd_create` is `SYS_timerfd_create` in libcs for android

**Comment by kalmari246** [comment]
yeh as a tmp fix i just added it to posix.rs https://github.com/kalmari246/async-timer-tmp-fix/blob/master/src/oneshot/posix.rs#L41 lol

**Comment by DoumanAsh** [comment]
@kalmari246 Actually, `timerfd` is not related to common POSIX implementation.
You should update it in https://github.com/kalmari246/async-timer-tmp-fix/blob/master/src/oneshot/timer_fd.rs

You could probably use posix timers for now, once timerfd gets fixed, you can enable tokio based timers again

**Comment by DoumanAsh** [comment]
Oh nevermind, you use posix's definitions in timerfd timer :D

If you want, you can make PR and I'll publish tmp fix.
But please move these definitions to timerfd (because they are used only there)

**Comment by kalmari246** [comment]
aight moved em, ig ill make a pr

**Comment by DoumanAsh** [comment]
@kalmari246 published `0.6.2` with your fix

**Comment by kalmari246** [comment]
hot
thanks
